ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/components/stack/StackComponentView.kt

```diff
@@ -142,6 +160,164 @@ private fun StackWithOverlaidBadge(
     }
 }
 
+/**
+ * @param topBadge Whether the badge should be placed on the top or bottom of the stack.
+ */
+@Suppress("LongMethod", "LongParameterList", "CyclomaticComplexMethod")
+@Composable
+private fun StackWithEdgeToEdgeBadge(
+    stackState: StackComponentState,
+    state: PaywallState.Loaded.Components,
+    badgeStack: StackComponentStyle,
+    topBadge: Boolean,
+    clickHandler: suspend (PaywallAction) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    SubcomposeLayout(
+        modifier = modifier,
+    ) { constraints ->
+        // Subcompose and measure the stack
+        val stackMeasurable = subcompose("stack") {
+            MainStackComponent(
+                stackState,
+                state,
+                clickHandler,
+            )
+        }.first()
+        val stackPlaceable = stackMeasurable.measure(constraints)
+
+        // Subcompose and measure the badge
+        val badgeMeasurable = subcompose("badge") {
+            StackComponentView(
+                // The background and border is applied to the parent container so we null it out here.
+                // We make the badge use all the available width without increasing the size of the main content.
+                badgeStack.copy(
+                    backgroundColor = null,
+                    size = Size(width = Fill, height = badgeStack.size.height),
+                    border = null,
+                    margin = PaddingValues(0.dp),
+                ),
+                state,
+                clickHandler,
+            )
+        }.first()
+        val badgePlaceable = badgeMeasurable.measure(constraints)
+        val badgeHeight = badgePlaceable.height
+
+        val backgroundCornerRadiusExtension = when (stackState.rcShape) {
+            is Shape.Pill -> stackPlaceable.height / 2
+            is Shape.Rectangle -> {
+                (
+                    (stackState.rcShape as? Shape.Rectangle)?.let { rcShape ->
+                        if (topBadge) {
+                            maxOf(rcShape.corners?.topLeading ?: 0.0, rcShape.corners?.topTrailing ?: 0.0)
+                        } else {
+                            maxOf(rcShape.corners?.bottomLeading ?: 0.0, rcShape.corners?.bottomTrailing ?: 0.0)
+                        }
+                    } ?: 0.0
+                    ).dp.toPx().roundToInt()
```

Comment:
Minor, more of an FYI:

```suggestion
                    ).dp.roundToPx()
```